website/static/js/share/index.js

```diff
@@ -102,15 +116,39 @@ ShareApp.controller = function() {
     });
 };
 
-var renderSort = function(ctrl){
+var SortBox = {
+    view: function(ctrl, params) {
+        var vm = params.vm;
         return m('.btn-group.pull-right', [
             m('button.btn.btn-default.dropdown-toggle', {
                     'data-toggle': 'dropdown',
                     'aria-expanded': 'false'
-                }, ['Sort by: ' + ctrl.vm.sort() + ' ', m('span.caret')]
+                }, ['Sort by: ' + vm.sort() + ' ', m('span.caret')]
             ),
-                m('ul.dropdown-menu', {'role': 'menu'}, ctrl.sideBarController.renderSort())]);
-    };
+                m('ul.dropdown-menu', {'role': 'menu'},
+                    $.map(Object.keys(vm.sortMap), function(a) {
```

Comment:
To maintain component identity if/when vm.sortMap is modified, you may need to pass `key` to the SortItems: https://lhorie.github.io/mithril/mithril.html#dealing-with-focus